bitnami/common/templates/_secrets.tpl

```diff
@@ -55,3 +55,48 @@ Params:
 
 {{- printf "%s" $key -}}
 {{- end -}}
+
+{{/*
+Generate secret password or retrieve one if already created.
+
+Usage:
+{{ include "common.secrets.manage.password" (dict "secret" "secret-name" "key" "keyName" "customValues" (list .Values.path.to.the.password1 .Values.path.to.the.password2) "length" "password-length" "strong" false "context" $) }}
+
+Params:
+  - secret - String - Required - Name of the 'Secret' resource where the password is stored.
+  - key - String - Required - Name of the key in the secret.
+  - customValues - Optional - List of parameters that can be used by the user to define the password. Will pick first parameter with a defined value.
+  - length - int - Optional - Length of the generated random password
+  - strong - Boolean - Optional - Whether to add "_" symbols to the generated random password
+  - context - Context - Required - Parent context.
+*/}}
+{{- define "common.secrets.manage.password" -}}
+
+{{- $password := "" }}
+{{- $providedPassword := "" }}
+{{- $passwordLength := default 10 .length}}
+{{- if .customValues }}
+  {{- $passwordList := compact .customValues }}
+  {{- $providedPassword = first $passwordList }}
+{{- end }}
+{{- $secret := (lookup "v1" "Secret" $.context.Release.Namespace .secret) }}
+
+{{- if $secret }}
+  {{- if index $secret.data .key }}
+  {{- $password = index $secret.data .key }}
+  {{- end -}}
+{{- else if $providedPassword }}
+  {{- $password = $providedPassword | b64enc | quote }}
+{{- else }}
+  {{- if .strong }}
+    {{- $subStr := list (lower (randAlpha 1)) (randNumeric 1) (upper (randAlpha 1)) | join "_" }}
```

Comment:
Could you show an example of what this part of the function does? until `cat` function.

Reply:
If we want a "strong" password of length `10`, the code would follow this steps:

1. We create a String `subStr` that will have 1 random lowercase letter, 1 number and 1 uppercase letter with the structure `a_1_A`. This is to pass the OrangeHRM password validation a cited in the description.
2. We generate a random string `password` of length 10 that can contain every printable ASCII character.
3. We replace every non letter or number character on `password` and change them for `@`. This is to avoid posible conflicts if the generated strings has characters like `"` or spaces. Then, we only save in the variable the generated string starting at the position `5`. That is, we don't save the characters `0-4`.
4. We concatenate `password` and `subStr`, which would give us a password of length `10`. Then we shuffle the characters to avoid a fixed pattern.

1. `d_4_Z`
2. `59o%p"9KLs`
3. `59o@p@9KLs` -> `@9KLs`
4. `d_4_Z@9KLs` -> `_sK94Z_@Ld`

Reply:
Perfect!